x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java

```diff
@@ -212,24 +223,33 @@ public void users(ActionListener<Collection<User>> listener) {
     }
 
 
-    private void getUserInfo(final String username, ActionListener<ReservedUserInfo> listener) {
+    private void getUserInfo(final String username, @Nullable SecureString credentials, ActionListener<ReservedUserInfo> listener) {
         if (securityIndex.indexExists() == false) {
-            listener.onResponse(getDefaultUserInfo(username));
+            getAutoconfiguredOrDefaultUser(username, credentials, listener);
         } else {
             nativeUsersStore.getReservedUserInfo(username, ActionListener.wrap((userInfo) -> {
                 if (userInfo == null) {
-                    listener.onResponse(getDefaultUserInfo(username));
+                    getAutoconfiguredOrDefaultUser(username, credentials, listener);
                 } else {
                     listener.onResponse(userInfo);
                 }
             }, (e) -> {
                 logger.error((Supplier<?>) () ->
-                        new ParameterizedMessage("failed to retrieve password hash for reserved user [{}]", username), e);
+                    new ParameterizedMessage("failed to retrieve password hash for reserved user [{}]", username), e);
                 listener.onResponse(null);
             }));
         }
     }
 
+    private void getAutoconfiguredOrDefaultUser(final String username, SecureString credentials,
+                                                ActionListener<ReservedUserInfo> listener) {
+        if (autoconfigured && username.equals(ElasticUser.NAME) && bootstrapUserInfo.verifyPassword(credentials)) {
```

Comment:
This condition here makes it so that the auto configured elastic password is actually checked twice (second time in ReservedRealm#doAuthenticate) when wrong.
Not a big issue in practice, but it shows that the control flow is not polished.